.github/workflows/build_and_deploy.yaml

```
@@ -0,0 +1,41 @@
+name: Solid POC and build and deploy
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  deploy:
+    name: Deploy POC app to Fargate
+    runs-on: ubuntu-latest
+    timeout-minutes: 60
+    permissions:
+      id-token: write
+      contents: read
+    steps:
+      - name: Set up SAM cli
```

```suggestion
      - name: Checkout
        uses: actions/checkout@v3

      - name: Set up SAM cli
```

Pretty sure you'll also need a checkout action first to get the latest branch contents